tools/src/main/scala/scala/scalanative/linker/Path.scala

```diff
@@ -14,6 +14,12 @@ sealed trait Path {
 
   /** Load given global and info about its dependencies. */
   def load(name: Global): Option[(Seq[Dep], Seq[Attr.Link], Defn)]
+
+  /** Dispose of given nir path. */
+  def close(): Unit
```

Comment:
See https://github.com/scala-native/scala-native/pull/416

Only virtual directories are resource-like, and those are managed directly by requiring implicit scope. Path should not have a close method.